WooCommerce/Classes/POS/Controllers/PointOfSaleItemsController.swift

```diff
@@ -79,12 +82,33 @@ class PointOfSaleItemsController: PointOfSaleItemsControllerProtocol {
         }
     }
 
+    @MainActor
+    func loadInitialChildItems(for parent: POSItem) async {
+        guard case let .parentProduct(parentProduct) = parent else {
+            return
+        }
+
+        itemsViewStateSubject.send(itemsViewStateSubject.value.copy(itemsStack: itemsViewStateSubject.value.itemsStack.copy(itemStates: [parent: .loading([])])))
```

Comment:
This adds a bug – if you go in to a new, unloaded parent, it will clear all the other saved itemStates for other parents.

E.g. 
1. Open a variable product, wait for it to load.
2. Go back to the root.
3. Open the same variable product: shows results immediately.
4. Go back to the root.
5. Open a new variable product: loads results
6. Go back to the root.
7. Open the variable product from step 1: loads results.

Reply:
Great catch! I was able to reproduce this, I updated it based on [your suggestion](https://github.com/woocommerce/woocommerce-ios/pull/14786#discussion_r1904042473) in https://github.com/woocommerce/woocommerce-ios/pull/14786/commits/ecf1a776db125363c49b93e56ec8de18662ecb43 https://github.com/woocommerce/woocommerce-ios/pull/14786/commits/15afbaed88c35108483e677cbca0ae24846bb3e9.